moulinette/authenticators/__init__.py

```
@@ -87,44 +83,57 @@ def __call__(self, password=None, token=None):
         """
         if self.is_authenticated:
             return self
-        store_session = True if password and token else False
 
-        if token:
+        #
+        # Authenticate using the password
+        #
+        if password:
             try:
-                # Extract id and hash from token
-                s_id, s_hash = token
-            except TypeError as e:
-                logger.error("unable to extract token parts from '%s' because '%s'", token, e)
-                if password is None:
-                    raise MoulinetteError('error_see_log')
-
-                logger.info("session will not be stored")
-                store_session = False
-            else:
-                if password is None:
-                    # Retrieve session
-                    password = self._retrieve_session(s_id, s_hash)
-
-        try:
-            # Attempt to authenticate
-            self.authenticate(password)
-        except MoulinetteError:
-            raise
-        except Exception as e:
-            logger.exception("authentication (name: '%s', vendor: '%s') fails because '%s'",
-                             self.name, self.vendor, e)
-            raise MoulinetteError('unable_authenticate')
-
-        # Store session
-        if store_session:
+                # Attempt to authenticate
+                self.authenticate(password)
+            except MoulinetteError:
+                raise
+            except Exception as e:
+                logger.exception("authentication (name: '%s', vendor: '%s') fails because '%s'",
+                                 self.name, self.vendor, e)
+                raise MoulinetteError('unable_authenticate')
+
+            self.is_authenticated = True
+
+            # Store session for later using the provided (new) token if any
+            if token:
```

We can loop forever without being able to store the session token and keep being asked to authenticate? Should we also raise an exception here to stop such a loop? Unsure if this can happen but trying to follow along all the same :man_technologist:

---

Eeeeh sorr I'm answering like 4 months later but can you elaborate on what loop you're thinking about ?